### PR TITLE
Remove extern declarations of OPENSSL_ia32cap_P

### PR DESCRIPTION
**Pull request by bernd-edlinger**
Use the header file internal/cryptlib.h instead.

### CODE REVIEW

File: crypto/whrlpool/wp_block.c [hunk 1]
```diff
@@ -75,7 +75,6 @@ typedef unsigned long long u64;
 #   define OPENSSL_SMALL_FOOTPRINT
 #  endif
 #  define GO_FOR_MMX(ctx,inp,num)     do {                    \
-        extern unsigned long OPENSSL_ia32cap_P[];               \
```
Comment (slontis) [inline crypto/whrlpool/wp_block.c]: Does this file need cryptlib.h included also?

Comment (bernd-edlinger) [reply crypto/whrlpool/wp_block.c]: I did my best to test this, but I'll double-check.

Comment (slontis) [reply crypto/whrlpool/wp_block.c]: looks like the travis picked this up

File: crypto/cryptlib.c [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 extern unsigned int OPENSSL_ia32cap_P[4];
 
-# if defined(OPENSSL_CPUID_OBJ) && !defined(OPENSSL_NO_ASM) && !defined(I386_ONLY)
+# if defined(OPENSSL_CPUID_OBJ)
```
Comment (slontis) [inline crypto/cryptlib.c]: Why are these removed - maybe a line in the commit?

Comment (bernd-edlinger) [reply crypto/cryptlib.c]: Ok will do.

All other files use the ia32cap_P if defined(OPENSSL_CPUID_OBJ)
but when no-asm or 386 is used to configure the CPUID_OBJ is not defined.
OTOH if the OPENSSL_CPUID_OBJ is linked it it outght to be initialized no
matter what other defines say.

File: crypto/whrlpool/wp_block.c (hunk 1 above)
Comment (bernd-edlinger) [reply crypto/whrlpool/wp_block.c]: ah, good, then. I was concerned about what happens when ONLY_I386 is defined.
and the variable would exist but initialization not happening.

Comment (richsalz) [reply crypto/whrlpool/wp_block.c]: #9582

**Comment by bernd-edlinger** [comment]
added include to wp_block.c and updated commit message.

**Comment by kroeckx** [comment]
Any reason why you didn't remove it in test/rdrand_sanitytest.c?

**Comment by kroeckx** [comment]
Should cryptlib.h have an `#if defined(OPENSSL_CPUID_OBJ)`?

**Comment by kroeckx** [comment]
I guess it's not OPENSSL_CPUID_OBJ, but:
```
#if     defined(__i386)   || defined(__i386__)   || defined(_M_IX86) || \
        defined(__x86_64) || defined(__x86_64__) || \
        defined(_M_AMD64) || defined(_M_X64)
```

**Comment by bernd-edlinger** [comment]
Yes, I was also thinking about that.

**Comment by bernd-edlinger** [comment]
> test/rdrand_sanitytest.c?

I was not sure if it is allowed to include internal header files, but probably it is okay.

File: crypto/cryptlib.c (hunk 2 above)
Comment (kroeckx) [reply crypto/cryptlib.c]: I think your commit message is confusing. OPENSSL_ia32cap_P always exists on x86/x86_64. OPENSSL_CPUID_OBJ just indicates that it's defined in some assembler file and that cryptlib.c shouldn't define it. It's a global variable, and so is initialized to 0. So I think most things that check for OPENSSL_CPUID_OBJ can just remove that check.

Comment (bernd-edlinger) [reply crypto/cryptlib.c]: Maybe, but checking a OPENSSL_ia32cap_P that is not initialized is pointless.
Nevertheless **this** ifdef cannot be removed.

Comment (kroeckx) [reply crypto/cryptlib.c]: I wasn't suggesting that this ifdef can be removed. (But the extern unsigned int just above it can.)

Comment (bernd-edlinger) [reply crypto/cryptlib.c]: It is likely that the header file is not included here, but that can be another PR.

**Comment by bernd-edlinger** [comment]
Merged to master as 24fd854.  Thanks!
